packages/apollo-federation/src/service/__tests__/buildFederatedSchema.test.ts

```diff
@@ -10,40 +17,88 @@ const EMPTY_DOCUMENT = {
   definitions: [],
 };
 
-describe('buildFederatedSchema', () => {
-  it(`should mark a type with a key field as an entity`, () => {
-    const schema = buildFederatedSchema(gql`
-      type Product @key(fields: "upc") {
-        upc: String!
-        name: String
-        price: Int
-      }
-    `);
+const testFederatedSchema = (
+  name: string,
+  createSchema: () => GraphQLSchema,
+  runTest: (schema: GraphQLSchema) => Promise<void>,
+) => {
+  const schema = createSchema();
+
+  it(name, async () => {
+    await runTest(schema);
+  });
+
+  it(`${name} (using "schema" argument)`, async () => {
+    await runTest(buildFederatedSchema(schema));
+  });
+};
 
-    expect(schema.getType('Product')).toMatchInlineSnapshot(`
+describe('buildFederatedSchema', () => {
+  testFederatedSchema(
+    'should mark a type with a key field as an entity',
+    () => {
+      return buildFederatedSchema(gql`
+        type Product @key(fields: "upc") {
+          upc: String!
+          name: String
+          price: Int
+        }
+      `);
+    },
+    async (schema: GraphQLSchema) => {
+      expect(schema.getType('Product')).toMatchInlineSnapshot(`
 type Product {
   upc: String!
   name: String
   price: Int
 }
 `);
 
-    expect(schema.getType('_Entity')).toMatchInlineSnapshot(
-      `union _Entity = Product`,
-    );
-  });
-
-  it(`should mark a type with multiple key fields as an entity`, () => {
-    const schema = buildFederatedSchema(gql`
-      type Product @key(fields: "upc") @key(fields: "sku") {
-        upc: String!
-        sku: String!
-        name: String
-        price: Int
-      }
-    `);
+      expect(schema.getType('_Entity')).toMatchInlineSnapshot(
+        `union _Entity = Product`,
+      );
+    },
+  );
+  testFederatedSchema(
+    `should mark a type with a key field as an entity`,
+    () => {
+      return buildFederatedSchema(gql`
+        type Product @key(fields: "upc") {
+          upc: String!
+          name: String
+          price: Int
+        }
+      `);
+    },
+    async (schema: GraphQLSchema) => {
+      expect(schema.getType('Product')).toMatchInlineSnapshot(`
+type Product {
+  upc: String!
+  name: String
+  price: Int
+}
+`);
 
-    expect(schema.getType('Product')).toMatchInlineSnapshot(`
+      expect(schema.getType('_Entity')).toMatchInlineSnapshot(
+        `union _Entity = Product`,
+      );
+    },
+  );
+
+  testFederatedSchema(
+    `should mark a type with multiple key fields as an entity`,
+    () => {
```

Comment:
For all these thunks, using 

```
() => buildFederatedSchema(...)
```
is a bit cleaner syntax. But as in the other comment, not sure why these need to be thunks in the first place.

Reply:
This definitely does not need to be a thunk.  PR coming soon.